### PR TITLE
Add more logging related to detection of PEs mapped anonymously by Wine

### PR DESCRIPTION
**Pull request by dpallotti**
It turns out that knowing which "failure" has happened is very useful...
In particular, distinguish the various error cases in `LinuxMap.cpp`, and add
logging for some of the error cases to `libunwindstack`'s
`MapInfo::GetFileMemoryFromAnonExecMapIfPeCoffTextSection` (I changed my mind
compared to https://github.com/google/orbit/pull/3590).

Test: Tried on a complex Silenus game.

### CODE REVIEW

File: third_party/libunwindstack/MapInfo.cpp [hunk 1]
```diff
@@ -271,10 +272,17 @@ Memory* MapInfo::GetFileMemoryFromAnonExecMapIfPeCoffTextSection() {
     return nullptr;
   }
 
+  Log::Info("Trying if anonymous executable map at %#lx-%#lx belongs to \"%s\"", start(), end(),
+            prev_name.c_str());
+  std::string error_message = android::base::StringPrintf(
```
Comment (florian-kuebler) [inline third_party/libunwindstack/MapInfo.cpp:277]: Nit/Optional: const?

File: src/ObjectUtils/LinuxMap.cpp [hunk 2]
```diff
@@ -149,30 +149,31 @@ class FileMappedIntoMemory {
       return false;
     }
 
-    ORBIT_LOG("Trying if executable map at %#x-%#x belongs to \"%s\"", map_start, map_end,
+    ORBIT_LOG("Trying if anonymous executable map at %#x-%#x belongs to \"%s\"", map_start, map_end,
               file_path_);
-    std::string error_message = absl::StrFormat(
-        "Executable map at %#x-%#x does NOT belong to \"%s\"", map_start, map_end, file_path_);
+    std::string error_message =
```
Comment (florian-kuebler) [inline src/ObjectUtils/LinuxMap.cpp:154]: Nit/optional: const?

**Review by florian-kuebler — APPROVED** [review]
LGTM. Sorry for the late review.

File: third_party/libunwindstack/MapInfo.cpp (hunk 1 above)
Comment (dpallotti) [reply third_party/libunwindstack/MapInfo.cpp]: Done.

File: src/ObjectUtils/LinuxMap.cpp (hunk 2 above)
Comment (dpallotti) [reply src/ObjectUtils/LinuxMap.cpp]: Done.

**Pull request merged**